aiida_quantumespresso/common/workchain/base/restart.py

```diff
@@ -77,136 +74,124 @@ def _load_error_handlers(self):
             for entry_point_name in get_entry_point_names(self._error_handler_entry_point):
                 try:
                     load_entry_point(self._error_handler_entry_point, entry_point_name)
-                    self.logger.info("loaded the '{}' entry point for the '{}' error handlers category"
-                        .format(entry_point_name, self._error_handler_entry_point))
-                except EntryPointError as exception:
-                    self.logger.warning("failed to load the '{}' entry point for the '{}' error handlers: {}"
-                        .format(entry_point_name, self._error_handler_entry_point, exception))
+                    self.logger.info("loaded the '%s' entry point for the '%s' error handlers category",
+                                     entry_point_name, self._error_handler_entry_point)
+                except exceptions.EntryPointError as exception:
+                    self.logger.warning("failed to load the '%s' entry point for the '%s' error handlers: %s",
+                                        entry_point_name, self._error_handler_entry_point, exception)
 
     @classmethod
     def define(cls, spec):
+        # yapf: disable
+        # pylint: disable=bad-continuation
         super(BaseRestartWorkChain, cls).define(spec)
         spec.input('max_iterations', valid_type=orm.Int, default=orm.Int(5),
-            help='maximum number of iterations the workchain will restart the calculation to finish successfully')
+            help='Maximum number of iterations the work chain will restart the calculation to finish successfully.')
         spec.input('clean_workdir', valid_type=orm.Bool, default=orm.Bool(False),
-            help='if True, work directories of all called calculation will be cleaned at the end of execution')
-        spec.exit_code(100, 'ERROR_ITERATION_RETURNED_NO_CALCULATION',
-            message='the run_calculation step did not successfully add a calculation node to the context')
+            help='If `True`, work directories of all called calculation will be cleaned at the end of execution.')
         spec.exit_code(101, 'ERROR_MAXIMUM_ITERATIONS_EXCEEDED',
-            message='the maximum number of iterations was exceeded')
-        spec.exit_code(102, 'ERROR_UNEXPECTED_CALCULATION_STATE',
-            message='the calculation finished with an unexpected calculation state')
-        spec.exit_code(103, 'ERROR_SECOND_CONSECUTIVE_SUBMISSION_FAILURE',
-            message='the calculation failed to submit, twice in a row')
-        spec.exit_code(104, 'ERROR_SECOND_CONSECUTIVE_UNHANDLED_FAILURE',
-            message='the calculation failed for an unknown reason, twice in a row')
+            message='The maximum number of iterations was exceeded.')
+        spec.exit_code(102, 'ERROR_SECOND_CONSECUTIVE_UNHANDLED_FAILURE',
+            message='The calculation failed for an unknown reason, twice in a row.')
```

Comment:
Absolutely agree. But while we are figuring this out we need to have the liberty to change these. That's also why I am changing most of this in the separate `3dd` branch. Ideally once I have a working situation, I discuss the proposal with other plugin developers to see if they can map the same guidelines on their work chains. Ideally input/output names and exit codes follow similar guidelines. Of course this will take time and combined with the idea that we don't want to change these lightly once released, this will push an official release far in the future. Not sure what the best balance will be here